sys-fs/bcachefs-tools/bcachefs-tools-1.4.0.ebuild

```diff
@@ -0,0 +1,260 @@
+
+EAPI=8
+
+CRATES="
+	aho-corasick@1.1.2
+	anstream@0.3.2
+	anstyle-parse@0.2.1
+	anstyle-query@1.0.0
+	anstyle-wincon@1.0.2
+	anstyle@1.0.2
+	anyhow@1.0.75
+	atty@0.2.14
+	autocfg@1.1.0
+	bitfield@0.14.0
+	bitflags@1.3.2
+	bitflags@2.4.1
+	byteorder@1.5.0
+	cc@1.0.83
+	cexpr@0.6.0
+	cfg-if@1.0.0
+	chrono@0.4.31
+	clang-sys@1.6.1
+	clap_builder@4.3.24
+	clap_complete@4.3.2
+	clap_derive@4.3.12
+	clap_lex@0.5.0
+	clap@4.3.24
+	colorchoice@1.0.0
+	colored@2.0.4
+	either@1.9.0
+	errno-dragonfly@0.1.2
+	errno@0.2.8
+	errno@0.3.7
+	fastrand@2.0.1
+	filedescriptor@0.8.2
+	gag@1.0.0
+	getset@0.1.2
+	glob@0.3.1
+	heck@0.4.1
+	hermit-abi@0.1.19
+	hermit-abi@0.3.3
+	io-lifetimes@1.0.11
+	is-terminal@0.4.9
+	itertools@0.9.0
+	lazy_static@1.4.0
+	lazycell@1.3.0
+	libc@0.2.150
+	libudev-sys@0.1.4
+	linux-raw-sys@0.3.8
+	linux-raw-sys@0.4.11
+	log@0.4.20
+	memchr@2.6.4
+	memoffset@0.8.0
+	minimal-lexical@0.2.1
+	nom@7.1.3
+	num-traits@0.2.17
+	once_cell@1.18.0
+	parse-display-derive@0.1.2
+	parse-display@0.1.2
+	paste@1.0.14
+	peeking_take_while@0.1.2
+	pkg-config@0.3.27
+	proc-macro-error-attr@1.0.4
+	proc-macro-error@1.0.4
+	proc-macro2@1.0.69
+	quote@1.0.33
+	redox_syscall@0.4.1
+	regex-automata@0.4.3
+	regex-syntax@0.6.29
+	regex-syntax@0.8.2
+	regex@1.10.2
+	rpassword@4.0.5
+	rustc-hash@1.1.0
+	rustix@0.37.27
+	rustix@0.38.25
+	shlex@1.2.0
+	strsim@0.10.0
+	syn@1.0.109
+	syn@2.0.39
+	tempfile@3.8.1
+	terminal_size@0.2.6
+	thiserror-impl@1.0.50
+	thiserror@1.0.50
+	udev@0.7.0
+	unicode-ident@1.0.12
+	utf8parse@0.2.1
+	uuid@1.6.1
+	version_check@0.9.4
+	winapi-i686-pc-windows-gnu@0.4.0
+	winapi-x86_64-pc-windows-gnu@0.4.0
+	winapi@0.3.9
+	windows_aarch64_gnullvm@0.48.5
+	windows_aarch64_msvc@0.48.5
+	windows_i686_gnu@0.48.5
+	windows_i686_msvc@0.48.5
+	windows_x86_64_gnu@0.48.5
+	windows_x86_64_gnullvm@0.48.5
+	windows_x86_64_msvc@0.48.5
+	windows-sys@0.48.0
+	windows-targets@0.48.5
+"
+
+# Upstream have a fork of bindgen and use cgit
+declare -A GIT_CRATES=(
+	[bindgen]="https://gitlab.com/Matt.Jolly/rust-bindgen-bcachefs;f773267b090bf16b9e8375fcbdcd8ba5e88806a8;rust-bindgen-bcachefs-%commit%/bindgen"
+)
+
+LLVM_MAX_SLOT=17
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit cargo flag-o-matic llvm multiprocessing python-any-r1 toolchain-funcs unpacker
+
+DESCRIPTION="Tools for bcachefs"
+HOMEPAGE="https://bcachefs.org/"
+if [[ ${PV} == "9999" ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://evilpiepirate.org/git/bcachefs-tools.git"
+else
+	SRC_URI="https://github.com/koverstreet/bcachefs-tools/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+		${CARGO_CRATE_URIS}"
+	S="${WORKDIR}/${P}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Apache-2.0 BSD GPL-2 MIT"
+SLOT="0"
+IUSE="fuse test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	fuse? ( >=sys-fs/fuse-3.7.0 )
+	app-arch/lz4:=
+	dev-libs/libaio
+	dev-libs/libsodium:=
+	dev-libs/userspace-rcu:=
+	sys-apps/keyutils
+	sys-apps/util-linux
+	sys-libs/zlib
+	virtual/udev
+"
```

Comment:
Sort unconditional deps before conditional ones. Also:
```
 * VDB: detected possibly incorrect RDEPEND (sys-fs/bcachefs-tools-1.4.0)
 *                   > app-arch/zstd:=
 * sys-apps/keyutils | sys-apps/keyutils:=
```

Reply:
Ack. I'll check if we're expecting a new release shortly and catch it then, else I'll look into it in a few days when I have time to compute. Thanks!